src/ctypes/ldouble_stubs.c

```diff
@@ -0,0 +1,525 @@
+    CAMLreturn(ctypes_copy_ldouble( OP (ldouble_custom_val(a), ldouble_custom_val(b)))); \
+  }
+
+FN2(powl)
+FN1(sqrtl)
+FN1(expl)
+FN1(logl)
+FN1(log10l)
+FN1(expm1l)
+FN1(log1pl)
+FN1(cosl)
+FN1(sinl)
+FN1(tanl)
+FN1(acosl)
+FN1(asinl)
+FN1(atanl)
+FN2(atan2l)
+FN2(hypotl)
+FN1(coshl)
+FN1(sinhl)
+FN1(tanhl)
+FN1(acoshl)
+FN1(asinhl)
+FN1(atanhl)
+FN1(ceill)
+FN1(floorl)
+FN1(fabsl)
+FN2(remainderl)
+FN2(copysignl)
+
+#undef OP2
+#undef FN1
+#undef FN2
+
+CAMLprim value ctypes_ldouble_frexp(value v) {
+  CAMLparam1(v);
+  CAMLlocal2(r, rfv);
+  long double f = ldouble_custom_val(v);
+  int ri;
+  long double rf;
+  r = caml_alloc_tuple(2);
+  rf = frexpl(f, &ri);
+  rfv = ctypes_copy_ldouble(rf);
+  Store_field(r,0, rfv);
+  Store_field(r,1, Val_int(ri));
+  CAMLreturn(r);
+}
+
+CAMLprim value ctypes_ldouble_ldexp(value vf, value vi) {
+  CAMLparam2(vf, vi);
+  CAMLlocal1(r);
+  long double f = ldouble_custom_val(vf);
+  int i = Int_val(vi);
+  long double rf = ldexpl(f, i);
+  r = ctypes_copy_ldouble(rf);
+  CAMLreturn(r);
+}
+
+CAMLprim value ctypes_ldouble_modf(value v) {
+  CAMLparam1(v);
+  CAMLlocal1(r);
+  long double f = ldouble_custom_val(v);
+  long double rf2;
+  long double rf1 = modfl(f, &rf2);
+  r = caml_alloc_tuple(2);
+  Store_field(r, 0, ctypes_copy_ldouble(rf1));
+  Store_field(r, 1, ctypes_copy_ldouble(rf2));
+  CAMLreturn(r);
+}
+
+enum {
+  ml_FP_NORMAL = 0,
+  ml_FP_SUBNORMAL,
+  ml_FP_ZERO,
+  ml_FP_INFINITE,
+  ml_FP_NAN,
+};
+
+CAMLprim value ctypes_ldouble_classify(value v){
+  CAMLparam1(v);
+  CAMLlocal1(r);
+  long double f = ldouble_custom_val(v);
+  switch (fpclassify(f)){
+  case FP_NORMAL    : r = Val_int(ml_FP_NORMAL); break;
+  case FP_SUBNORMAL : r = Val_int(ml_FP_SUBNORMAL); break;
+  case FP_ZERO      : r = Val_int(ml_FP_ZERO); break;
+  case FP_INFINITE  : r = Val_int(ml_FP_INFINITE); break;
+  case FP_NAN       : 
+  default           : r = Val_int(ml_FP_NAN); break;
+  }
+  CAMLreturn(r);
+}
+
+static char *format_ldouble(char *fmt, long double d) {
+  static size_t buf_len = 10;
+  static char *buf = NULL;
+  static char *empty = "";
+  size_t print_len = 0;
+
+  // allocate initial string
+  if (buf == NULL) buf = malloc(buf_len);
+  if (buf == NULL) return empty; // oops...stuck
+
+  // try to print
+  print_len = snprintf(buf, buf_len, fmt, d);
+
+  // not enough space - reallocate and try again
+  if (print_len >= buf_len) {
+    // re-allocate string
+    if (buf) free(buf);
+    buf = malloc(print_len+1);
+    buf_len = print_len+1;
+    // try again
+    return format_ldouble(fmt, d);
+  } else 
+    // ok
+    return buf;
+}
+
+CAMLprim value ctypes_ldouble_format(value fmt, value d) {
+  CAMLparam2(fmt, d);
+  CAMLreturn(caml_copy_string(format_ldouble( String_val(fmt), ldouble_custom_val(d))));
+}
+
+CAMLprim value ctypes_ldouble_of_string(value v) {
+  CAMLparam1(v);
+  char *str = String_val(v);
+  char *end = str + caml_string_length(v);
+  long double r = strtold(str, &end);
```

Comment:
This needs to check that conversion succeeded (by checking that `end` does actually point to the end of the string after the call), and ideally raise `Invalid_argument` if it failed.